cds_ils/migrator/items/api.py

```diff
@@ -41,16 +41,18 @@ def set_internal_location_pid(record):
 
 
 def set_document_pid(record):
-    """Set document pid for item."""
+    """Set document pid for item and returns the document."""
     document_cls = current_app_ils.document_record_cls
 
     # find document
     record["document_pid"] = None
     try:
 
-        record["document_pid"] = get_record_by_legacy_recid(
+        document = get_record_by_legacy_recid(
```

Comment:
lines 51:70 could be in another function, see the comments below

and here simply
```
document = find_document_for_item(record)
record["document_pid"] = document.pid.pid_value
```